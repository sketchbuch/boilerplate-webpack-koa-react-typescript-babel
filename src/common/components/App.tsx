import React from 'react';

import '../styles/index.css';

interface AppProps {
  title: string;
}

const App = ({ title = 'Sketchbuch' }: AppProps): JSX.Element => {
  return (
    <div>
      <h1>{title}</h1>
    </div>
  );
};

export default App;
