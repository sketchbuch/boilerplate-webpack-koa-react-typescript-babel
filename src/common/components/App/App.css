.App__title {
  color: #292727;
  text-align: center;
}
