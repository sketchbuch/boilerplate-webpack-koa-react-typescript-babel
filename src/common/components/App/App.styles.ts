import styled from 'styled-components';

export const StyledApp = styled.div`
  text-align: center;
  font-size: 2rem;
`;

export const StyledP = styled.div`
  text-align: center;
  font-weight: bold;
  font-size: 1rem;
`;
