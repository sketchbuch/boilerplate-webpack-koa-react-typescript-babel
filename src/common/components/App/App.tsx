import React, { useEffect } from 'react';
import { APP_LOADED } from '../../constants/actions';
import { Props, Store } from './App.interface';
import { useSelector, useDispatch } from 'react-redux';
import { StyledApp, StyledP } from './App.styles';

const App = ({ title }: Props): JSX.Element => {
  const loaded: boolean = useSelector<Store, boolean>(state => state.app);
  const dispatch = useDispatch();

  useEffect(() => {
    const timer: number = window.setTimeout(
      () => dispatch({ type: APP_LOADED }),
      1000
    );

    return () => clearTimeout(timer);
  }, []);

  return (
    <StyledApp>
      <h1 data-testid="app-title">{title}</h1>
      {loaded === true ? (
        <StyledP data-testid="app-loaded">Loaded!</StyledP>
      ) : (
        <StyledP data-testid="app-loading">Loading...</StyledP>
      )}
    </StyledApp>
  );
};

export default App;
