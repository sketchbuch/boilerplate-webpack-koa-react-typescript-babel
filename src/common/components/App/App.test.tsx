import React from 'react';
import { wait } from '@testing-library/react';
import App, { AppProps } from './App';
import renderWithRedux from '../../tests/renderWithRedux';

describe('<App />', () => {
  const props: AppProps = {
    title: 'A headline',
  };
  const LOADING_TXT: string = 'Loading...';
  const LOADED_TXT: string = 'Loaded!';

  test('Renders the title', () => {
    const { container, getByText } = renderWithRedux(<App {...props} />);
    expect(container.querySelector('.App__title')).toBeInTheDocument();
    expect(getByText(props.title)).toBeInTheDocument();
  });

  test('Renders loading message', () => {
    const { container, getByText, queryByText } = renderWithRedux(
      <App {...props} />
    );
    expect(container.querySelector('#app-loading')).toBeInTheDocument();
    expect(container.querySelector('#app-loaded')).not.toBeInTheDocument();
    expect(getByText(LOADING_TXT)).toBeInTheDocument();
    expect(queryByText(LOADED_TXT)).not.toBeInTheDocument();
  });

  test('Renders loaded message after 1 second', async () => {
    const { container, getByText, queryByText } = renderWithRedux(
      <App {...props} />
    );
    await wait(
      () => {
        expect(container.querySelector('#app-loading')).not.toBeInTheDocument();
        expect(container.querySelector('#app-loaded')).toBeInTheDocument();
        expect(queryByText(LOADING_TXT)).not.toBeInTheDocument();
        expect(getByText(LOADED_TXT)).toBeInTheDocument();
      },
      {
        timeout: 1000,
      }
    );
  });
});
