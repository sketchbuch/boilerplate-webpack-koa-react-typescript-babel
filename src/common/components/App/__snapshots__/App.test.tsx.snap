// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Styled Components: <StyledApp /> 1`] = `
.c0 {
  text-align: center;
  font-size: 2rem;
}

<div
  className="c0"
/>
`;

exports[`<App /> Styled Components: <StyledP /> 1`] = `
.c0 {
  text-align: center;
  font-weight: bold;
  font-size: 1rem;
}

<div
  className="c0"
/>
`;
